Scrapy-basics/scrapy_basics/scrapy_basics/spiders/scrapyfetch.py

```diff
@@ -0,0 +1,195 @@
+                yield scrapy.Request(url=url, callback=self.parse_page, dont_filter=True, meta=xpath_selector)
+
+            yield scrapy.Request(url=url, callback=xpath_selector['page_selector_function'], dont_filter=True,
+                                 meta=xpath_selector)
+
+    def parse_all_pages(self, response):
+        """
+        Extracts the links of all pages to scrape at once, and scrape them
+        :param response: the response of object of the request
+        """
+
+        pages = response.xpath(response.meta.get('pages_selector')).extract()
+
+        for page in pages:
+            next_page = response.urljoin(page)  # concat the sub-link to the main link to form a correct full link
+            yield scrapy.Request(url=next_page, callback=self.parse_page, meta=response.meta)
+
+    def parse_by_next_page(self, response):
+        """
+        Extracts the links of the pages to scrape by imitating the process of pressing "Next Page"
+        in the website, and scrape them
+        :param response: the response of object of the request
+
+        """
+
+        next_page = response.xpath(response.meta.get('pages_selector')).extract_first()  # extract next page link
+
+        if next_page:  # if this is the last page the recursive call will stop
+            next_page = response.urljoin(next_page)
+
+            # scrape the next page and its information to the csv file
+            yield scrapy.Request(url=next_page, callback=self.parse_page, meta=response.meta, dont_filter=True)
+
+            # recursively go to the next page to iterate more pages
+            yield scrapy.Request(url=next_page, callback=self.parse_by_next_page, meta=response.meta)
+
+    def parse_misc(self, response):
+        """
+        :param response: the response of object of the request
+        Scrape the contents of the product's page inside response and extract:
+        gpu_model, in_stock
+        """
+
+        # read the already fetched info from the meta data in the response object
+        gpu_name = response.meta.get('gpu_name')
+        url = response.meta.get('url')
+        gpu_price = response.meta.get('gpu_price')
+        fetch_ts = response.meta.get('fetch_ts')
+        store_name = response.meta.get('store_name')
+
+        in_stock = response.xpath(response.meta.get("in_stock_selector")).extract_first()
+
+        # the case of having the number of remaining items
+        if response.meta.get("in_stock_type") == "number":
+            if not in_stock:  # if no information was specified the item is considered out of stock
+                in_stock = False
+            else:
+                in_stock_text = in_stock.split(" ")
+                in_stock = False
+                for element in in_stock_text:
+                    try:
+                        if int(element) > 0:
+                            in_stock = True
+                            break
+                    except:
```

Comment:
Handling exceptions that way is not very good practice in Python. Could you please specify concrete exception/s in except clause, or at least add comment, why you just skipping all of them.